### PR TITLE
Fix golint in k8sclient

### PR DESCRIPTION
**Pull request by m1093782566**

### CODE REVIEW

**Comment by m1093782566** [comment]
@deepak-vij @shivramsrivastava

**Comment by m1093782566** [comment]
/assign @ICGog 
for approval.

File: pkg/k8sclient/keyed_queue.go [hunk 1]
```diff
@@ -20,16 +20,23 @@ import (
 	"sync"
 )
 
+// Queue is an interface which abstracts a queue.
 type Queue interface {
+	// Adds enqueues a key and its associated item.
```
Comment (shashidharatd) [inline pkg/k8sclient/keyed_queue.go]: s/Adds/Add

File: pkg/k8sclient/keyed_queue.go [hunk 2]
```diff
@@ -20,16 +20,23 @@ import (
...
 	Add(key interface{}, item interface{})
+	// Get removes an item from the queue.
```
Comment (shashidharatd) [inline pkg/k8sclient/keyed_queue.go]: Looks something odd that `Get` removes an item from the queue. is this true?

File: pkg/k8sclient/keyed_queue.go [hunk 3]
```diff
@@ -40,15 +47,17 @@ func NewKeyedQueue() *Type {
...
 	queue []tk
 	// Items to be processed associated with each key.
 	items map[tk][]interface{}
 	// Items to be queued once the keys are processed.
 	toQueue map[tk][]interface{}
-	// Set of keys curretly under processing.
-	processing   set
+	// Set of keys currently under processing.
+	processing set
+	// Flag representing if the queue is shutting down.
```
Comment (shashidharatd) [inline pkg/k8sclient/keyed_queue.go]: s/Flag/shuttingDown

File: pkg/k8sclient/keyed_queue.go [hunk 4]
```diff
@@ -79,19 +89,19 @@ func (q *Type) Add(key interface{}, item interface{}) {
...
-			q.cond.Signal()
-		}
+		// New key in the queue. Send signal.
+		q.items[key] = append(q.items[key], item)
+		q.queue = append(q.queue, key)
+		q.cond.Signal()
 	}
 }
 
+// Get removes an item from the queue and insets the item to the currently processing key set.
```
Comment (shashidharatd) [inline pkg/k8sclient/keyed_queue.go]: typo insets

File: pkg/k8sclient/nodewatcher.go [hunk 5]
```diff
@@ -2,7 +2,7 @@
 Copyright 2018 The Kubernetes Authors.
 
 Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
+you may not use nw file except in compliance with the License.
```
Comment (shashidharatd) [inline pkg/k8sclient/nodewatcher.go]: typo nw

File: pkg/k8sclient/nodewatcher.go [hunk 6]
```diff
@@ -34,6 +34,7 @@ import (
 	"k8s.io/poseidon/pkg/firmament"
 )
 
+// NewNodeWatcher initializes a NodeWatcher base on the given Kubernetes client and Firmament client.
```
Comment (shashidharatd) [inline pkg/k8sclient/nodewatcher.go]: s/base/based

File: pkg/k8sclient/podwatcher.go [hunk 7]
```diff
@@ -2,7 +2,7 @@
 Copyright 2018 The Kubernetes Authors.
 
 Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
+you may not use pw file except in compliance with the License.
```
Comment (shashidharatd) [inline pkg/k8sclient/podwatcher.go]: typo pw

File: pkg/k8sclient/keyed_queue.go (hunk 2 above)
Comment (m1093782566) [reply pkg/k8sclient/keyed_queue.go]: Hmmm...

```
Get removes an item from the queue and insets the item to the currently processing key set.
```

What about the new comment message?

File: pkg/k8sclient/nodewatcher.go (hunk 5 above)
Comment (m1093782566) [reply pkg/k8sclient/nodewatcher.go]: Oops... It's fixed now. Thanks!

File: pkg/k8sclient/keyed_queue.go (hunk 4 above)
Comment (m1093782566) [reply pkg/k8sclient/keyed_queue.go]: Done.

File: pkg/k8sclient/keyed_queue.go (hunk 3 above)
Comment (m1093782566) [reply pkg/k8sclient/keyed_queue.go]: Done

File: pkg/k8sclient/keyed_queue.go (hunk 1 above)
Comment (m1093782566) [reply pkg/k8sclient/keyed_queue.go]: Fixed.

File: pkg/k8sclient/nodewatcher.go (hunk 6 above)
Comment (m1093782566) [reply pkg/k8sclient/nodewatcher.go]: Fixed.

File: pkg/k8sclient/podwatcher.go (hunk 7 above)
Comment (m1093782566) [reply pkg/k8sclient/podwatcher.go]: Fixed now.

**Comment by m1093782566** [comment]
@shashidharatd 

Thanks for your diligently reviewing, all comments are addressed now. PTAL.

Thanks!

**Comment by shashidharatd** [comment]
/lgtm
Thanks @m1093782566

**Comment by m1093782566** [comment]
ping @ICGog @shivramsrivastava for approval :)

**Comment by m1093782566** [comment]
THANKS!

**Pull request merged**